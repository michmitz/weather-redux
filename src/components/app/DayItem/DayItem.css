.dayDiv {
  margin: 2em;
}

.date {
  color: rgb(255, 127, 123);
}