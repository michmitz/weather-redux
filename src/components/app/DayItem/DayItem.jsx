import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../styles/styles.css';
import dayStyles from './DayItem.css';

const DayItem = ({ date, currentTemp, hi, lo, description }) => (
  <div className={dayStyles.dayDiv} data-testid="day-div">
    <h2>{date}</h2>
    <h3>Current Temp: {currentTemp}</h3>
    <p>Hi: {hi}</p>
    <p>Lo: {lo}</p>
    <p>Description: {description}</p>
  </div> 
);

DayItem.propTypes = {
  date: PropTypes.string.isRequired,
  currentTemp: PropTypes.number.isRequired,
  hi: PropTypes.number.isRequired,
  lo: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired
};

export default DayItem;

