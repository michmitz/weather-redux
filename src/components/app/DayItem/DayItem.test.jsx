import React from 'react';
import { render, cleanup } from '@testing-library/react';
import DayItem from './DayItem';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('User component', () => {
  afterEach(() => cleanup());
  it('renders User', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <DayItem />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});