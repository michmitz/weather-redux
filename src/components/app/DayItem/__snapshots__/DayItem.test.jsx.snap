// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders DayItem 1`] = `
<DocumentFragment>
  <div
    data-testid="day-div"
  >
    <h2>
      Date:
    </h2>
    <h3>
      Current Temp: 
    </h3>
    <h3>
      Hi: 
    </h3>
    <h3>
      Lo: 
    </h3>
    <p />
  </div>
</DocumentFragment>
`;
